### PR TITLE
feat: add python implementation

### PR DESCRIPTION
**Pull request by dOrgJelli**
This PR adds http-plugin for the python client.
- Closes: polywrap/python-client#71

### CODE REVIEW

File: implementations/py/polywrap-http-plugin/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Polywrap HTTP Plugin
```
Comment (dOrgJelli) [inline implementations/py/polywrap-http-plugin/README.md:1]: Can we include more detail here?

File: implementations/py/polywrap-http-plugin/polywrap_http_plugin/__init__.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+from httpx import AsyncClient, Response
+from polywrap_plugin import PluginModule, PluginPackage
+from polywrap_core import Invoker
+from polywrap_result import Ok, Result
+from polywrap_manifest import WrapManifest
+from polywrap_msgpack.generic_map import GenericMap
+
+
+class HttpResponseType(Enum):
```
Comment (dOrgJelli) [inline implementations/py/polywrap-http-plugin/polywrap_http_plugin/__init__.py:15]: What's the status on getting `plugin/python` codegen integrated into the CLI?

File: implementations/py/polywrap-http-plugin/polywrap_http_plugin/__init__.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                status=res.status_code,
+                statusText=res.reason_phrase,
+                headers=GenericMap(dict(res.headers)),
+                body=res.text,
+            )
+        )
+
+
+def http_plugin():
+    manifest_path = Path(__file__).parent.joinpath("manifest.json")
```
Comment (dOrgJelli) [inline implementations/py/polywrap-http-plugin/polywrap_http_plugin/__init__.py:143]: Ideally we don't want to load this via the file-system, in cases where the python program doesn't allow access.

File: implementations/py/polywrap-http-plugin/tests/test_ipfs.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    )
+
+    config = PolywrapClientConfig(resolver=resolver)
+    client = PolywrapClient(config)
+
+    ipfs_wrapper_path = Path(__file__).parent.joinpath("ipfs-wrapper")
+    ipfs_wrapper_uri = Uri(f"fs/{ipfs_wrapper_path}")
+
+    result = await client.invoke(
+        InvokerOptions(
```
Comment (dOrgJelli) [inline implementations/py/polywrap-http-plugin/tests/test_ipfs.py:28]: Not sure why this test is here... doesn't seem relevant to the HTTP plugin.

File: implementations/py/polywrap-http-plugin/tests/test_plugin.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+from typing import cast
+from polywrap_http_plugin import http_plugin, HttpResponse
+from polywrap_client import PolywrapClient, PolywrapClientConfig
+from polywrap_core import Uri, IUriResolver, InvokerOptions
+from polywrap_uri_resolvers import StaticResolver, RecursiveResolver
+from polywrap_uri_resolvers.uri_resolver_aggregator import UriResolverAggregator
+from polywrap_uri_resolvers.legacy import FsUriResolver, SimpleFileReader
+
+
+async def test_plugin():
```
Comment (dOrgJelli) [inline implementations/py/polywrap-http-plugin/tests/test_plugin.py:11]: Can we please expand on the testing that's being done here to cover all public methods? Can use the JS tests as a reference: https://github.com/polywrap/http/blob/main/implementations/js/src/__tests__/e2e/e2e.spec.ts

**Review by dOrgJelli — COMMENTED** [review]
Also @Niraj-Kamdar I don't think the `__pycache__` directory & `typings/` directory should be included in source control, as they're build artifacts (if I'm not mistaken).

File: implementations/py/polywrap-http-plugin/polywrap_http_plugin/__init__.py (hunk 3 above)
Comment (Niraj-Kamdar) [reply implementations/py/polywrap-http-plugin/polywrap_http_plugin/__init__.py:143]: There will be parse wrap.info.py in the future like js when we add codegen for python plugin

Comment (dOrgJelli) [reply implementations/py/polywrap-http-plugin/polywrap_http_plugin/__init__.py:143]: Okay sounds good

File: implementations/js/package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
```
Comment (Niraj-Kamdar) [inline implementations/js/package.json:1]: TODO: fix this

**Comment by Niraj-Kamdar** [comment]
TODO: CI/CD

File: implementations/py/README.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+# @polywrap/http-plugin-js
```
Comment (dOrgJelli) [inline implementations/py/README.md:1]: This should be updated

File: implementations/py/schema.graphql [hunk 8]
```diff
@@ -0,0 +1,4 @@
+#import { Module } into Http from "ens/wraps.eth:http@1.1.0"
```
Comment (dOrgJelli) [inline implementations/py/schema.graphql:1]: As we discussed previously, this sort of "import interface into namespace + use the graphql `implements` keyword" is actually "interface extension" and not "implementation" as we want.

I'd recommend doing an un-namespaced import like so:
`#import * from "ens/wraps.eth:http@1.1.0"`

Comment (Niraj-Kamdar) [reply implementations/py/schema.graphql:1]: I was working on this but it's making CI red since it will require minor changes python bindings

File: .github/workflows/http-plugin-py-cd.yaml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+      - main
+    paths:
+      - "implementations/py/**"
+
+jobs:
+  cd:
+    name: http-plugin-py-cd
+    if: |
+      github.event.pull_request.merged &&
+      startsWith(github.event.pull_request.title, '[CD]') != true &&
```
Comment (cbrzn) [inline .github/workflows/http-plugin-py-cd.yaml:16]: shouldn't this be:
```suggestion
      startsWith(github.event.pull_request.title, '[CD]') == true &&
```
also, just wondering: do we want to have the same pattern as other implementations? Meaning that when we add `[CD]` in the PR we do a release of *all* implementations, or we would like to do this 1:1 (1 PR, 1 Release)

I don't have strong opinions but thought its worth sharing this :smile:

File: implementations/py/tests/integration/wrapper/wrap.wasm
Comment (cbrzn) [inline implementations/py/tests/integration/wrapper/wrap.wasm:1]: nit: shouldn't this be generated when the test runs?

Comment (Niraj-Kamdar) [reply implementations/py/tests/integration/wrapper/wrap.wasm:1]: It is going to be the same since it's a thin wrapper around the plugin. I don't want to copy the source code of this integration wrapper for every plugin implementation. We should probably change how we do integration testing in every plugin, but I believe that's out of the scope of the PR.
